vj4/handler/contest.py

```diff
@@ -53,13 +53,24 @@ def status_text(self, tdoc):
     else:
       return 'Done'
 
-  def can_show(self, tdoc):
-    return contest.RULES[tdoc['rule']].show_func(tdoc, self.now)
 
+class ContestVisibilityMixin(object):
+  def _can_view_hidden_status_scoreboard(self, tdoc):
+    return self.has_perm(builtin.PERM_VIEW_CONTEST_HIDDEN_STATUS_AND_SCOREBOARD)
```

Comment:
PERM_VIEW_CONTEST_HIDDEN_STATUS_AND_SCOREBOARD ???

You said `status` is renamed to `scoreboard`.